### DISCUSSION TITLE
Just a note about making it work

### DISCUSSION THREAD

**Discussion by OpenSorce1** [mathoudebine/turing-smart-screen-python#197] [General]
Hi! I'm running a fresh install of Ubuntu 22.04 and downloading everything and installing all the pre-reqs it still wouldn't work.

I was getting the [Errno 13] Permission denied: '/dev/ttyACM0' issue. Even after adding my account to dialout. 
I did sudo dmesg -w and unplugged then plugged in my screen to make sure it was on /dev/ttyACM0
So I did a 'sudo chmod ugo+rw /dev/ttyACM0' also tried 'sudo chmod 666 /dev/ttyACM0'
I noticed that when I would run python3 ./configure.py then 'Save and Run' the screen would flash then give the permission error.
I checked the permissions of /dev/ttyACM0 and they had reverted to the previous perms.
So, I added a file to /etc/udev/rules.d/ called 50-myusb.rules
In that file I put the following: KERNEL=="ttyACM[0-9]*",MODE="0666"
Now it works! Apparently it was reconnecting and then defaulting to the preset permissions when trying to launch.
I will likely find where it does this and fix that but this work-around works for now.
Hope it's helpful to someone else.

This fix should likely be considered more of a temporary work-around than a solution. It leaves your USB devices completely accessible by all users that connect to your PC. For a home user, this is not an issue. For a server, it's a no-no. I'll figure out what was setting the USB device to restricted access and fix that. When I do, I'll reply or edit this post :-)

Devs: Thanks so much for making this!

**Comment by SdPatSdP** [comment]
Me too.
Adding file to
/etc/udev/rules.d/ 

called
50-myusb.rules

In that file I put the following: 
KERNEL=="ttyACM[0-9]*",MODE="0666"

worked, but it looks unsafe to me too..

I have a pristine Debian 12 installation (with just update/upgrade)
Version of  turing-smart-screen-python is 3.5.0 latest as when I'm posting.

Great work to pin it out.
SdP

**Comment by ColinShorts** [comment]
Having a 3.5 and 9.2 screens, I used the following rules;
```
ACTION=="add", ATTRS{idVendor}=="1a86", ATTRS{idProduct}=="5722", SYMLINK+="turing35", MODE:="0660", OWNER="root", GROUP:="dialout"
SUBSYSTEM=="usb", ATTR{idVendor}=="1cbe", ATTR{idProduct}=="0092", MODE="0660", OWNER="root", GROUP="dialout"
```
In the first case I've created a symlink so we can specify a static "COM_PORT", in the second, I'm just picking it up as auto, and just setting the permissions on the device node directly.
